Sources/SwiftRefactor/CallToTrailingClosures.swift

```diff
@@ -0,0 +1,143 @@
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftBasicFormat
+import SwiftSyntax
+
+/// Convert a call with inline closures to one that uses trailing closure
+/// syntax. Returns `nil` if there's already trailing closures or there are no
+/// closures within the call. Pass `startAtArgument` to specify the argument
+/// index to start the conversion from, ie. to skip converting closures before
+/// `startAtArgument`.
+///
+/// ## Before
+/// ```
+/// someCall(closure1: { arg in
+///   return 1
+/// }, closure2: { arg in
+///   return 2
+/// })
+/// ```
+///
+/// ## After
+/// ```
+/// someCall { arg in
+///   return 1
+/// } closure2: { arg in
+///   return 2
+/// }
+/// ```
+public struct CallToTrailingClosures: SyntaxRefactoringProvider {
+  public struct Context {
+    public let startAtArgument: Int
+
+    public init(startAtArgument: Int = 0) {
+      self.startAtArgument = startAtArgument
+    }
+  }
+
+  // TODO: Rather than returning nil, we should consider throwing errors with
+  // appropriate messages instead.
+  public static func refactor(syntax call: FunctionCallExprSyntax, in context: Context = Context()) -> FunctionCallExprSyntax? {
+    return call.convertToTrailingClosures(from: context.startAtArgument)?.formatted().as(FunctionCallExprSyntax.self)
+  }
+}
+
+extension FunctionCallExprSyntax {
+  fileprivate func convertToTrailingClosures(from startAtArgument: Int) -> FunctionCallExprSyntax? {
+    guard trailingClosure == nil, additionalTrailingClosures == nil, leftParen != nil, rightParen != nil else {
+      // Already have trailing closures
+      return nil
+    }
+
+    var closures = [(original: TupleExprElementSyntax, closure: ClosureExprSyntax)]()
+    for arg in argumentList.dropFirst(startAtArgument) {
+      guard var closure = arg.expression.as(ClosureExprSyntax.self) else {
+        closures.removeAll()
+        continue
+      }
+
+      // Trailing comma won't exist any more, move its trivia to the end of
+      // the closure instead
+      if let comma = arg.trailingComma {
+        closure = closure.with(\.trailingTrivia, closure.trailingTrivia.merging(triviaOf: comma))
+      }
+      closures.append((arg, closure))
+    }
+
+    guard !closures.isEmpty else {
+      return nil
+    }
+
+    // First trailing closure won't have label/colon. Transfer their trivia.
+    var trailingClosure = closures.first!.closure
+      .with(\.leadingTrivia, Trivia()
+        .merging(triviaOf: closures.first!.original.label)
+        .merging(triviaOf: closures.first!.original.colon)
+        .merging(closures.first!.closure.leadingTrivia))
```

Comment:
Are you dropping the trivia of `closures.first!.trailingComma` or am I missing something?

Reply:
It's added above. FWIW I added quite a few more comments in the trivia test that hopefully catches all the cases now 😅. Worth checking those as well - IMO there's a few odd spaces that we should probably remove, but I think it's good enough for now.